tests/Unit/TestingFramework.hpp

```diff
@@ -212,6 +212,48 @@ struct check_iterable_approx<
     }
   }
 };
+
+#define CHECK_MATRIX_APPROX(a, b)                                            \
```

Comment:
To test numerical equality of matrices I added this macro. Could someone have a look at this? There are also a few comments in the code.

Reply:
Why not just use `check_iterable_approx`? Matrices should be iterable

Reply:
They are only row-iterable or column-iterable: https://bitbucket.org/blaze-lib/blaze/wiki/Matrix%20Operations#!iterators

Reply:
We shouldn't block on this but I've raised an issue with Blaze that their implementation is broken: https://bitbucket.org/blaze-lib/blaze/issues/178/matrix-begin-and-end-are-broken

Reply:
That's good, I agree they should adhere to the STL `begin()` and `end()`.